rcl_logging_spdlog/test/test_logging_interface.cpp

```diff
@@ -62,6 +65,42 @@ class RestoreEnvVar
   const std::string value_;
 };
 
+class TemporaryLogDir final
+{
+public:
+  TemporaryLogDir()
+  : orig_ros_log_dir_value_(rcpputils::get_env_var("ROS_LOG_DIR"))
+  {
+    // Create a random directory in our current path
+    std::random_device dev;
+    std::mt19937 prng(dev());
+    std::uniform_int_distribution<uint64_t> rand(0);
+
+    std::stringstream ss;
+    ss << std::hex << rand(prng);
+
+    local_log_dir_ = std::filesystem::current_path() / ss.str();
+
+    if (!std::filesystem::create_directories(local_log_dir_)) {
+      throw std::runtime_error("Failed to make temporary log directory");
+    }
+
+    rcpputils::set_env_var("ROS_LOG_DIR", local_log_dir_.string().c_str());
```

Comment:
> we can assume this always succeeds, right? because above it already created the `local_log_dir_`? i do not see any actual problem here but just checking.

Yes, we can assume it always succeeds, but that's because of a bug in the `rcpputils::set_env_var` API.  In particular, `rcpputils::set_env_var` either throws an exception on error, or returns `true` on success.  It is a really weird API, and should probably be fixed at some point.